compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -346,19 +346,24 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     );
                 }
 
-                if let Some(def) = actual.ty_adt_def() {
-                    if let Some(full_sp) = tcx.hir().span_if_local(def.did()) {
-                        let def_sp = tcx.sess.source_map().guess_head_span(full_sp);
-                        err.span_label(
-                            def_sp,
-                            format!(
-                                "{} `{}` not found {}",
-                                item_kind,
-                                item_name,
-                                if def.is_enum() && !is_method { "here" } else { "for this" }
-                            ),
-                        );
-                    }
+                let span_and_label_suffix = match actual.kind() {
+                    ty::Param(param_type) => self.in_progress_typeck_results.map(|table| {
+                        let table_owner = table.borrow().hir_owner;
+                        let generics = self.tcx.generics_of(table_owner.to_def_id());
```

Comment:
Can you do this instead:
```suggestion
                    ty::Param(param_type) => {
                        let generics = self.tcx.generics_of(tcx.hir().body_owner_def_id(self.body_id).to_def_id());
```

Reply:
Also if you can change the other usage of `in_progress_typeck_results` in this file, that would be helpful